apps/web/src/containers/TransactionScreen/components/data-section.tsx

```diff
@@ -0,0 +1,128 @@
+import { alpha, Box, styled, useTheme } from '@mui/material';
+import { Button, H2, LargeInputTextEmphasized } from '@notional-finance/mui';
+import { ExternalLinkIcon } from '@notional-finance/icons';
+import { TotalBox } from './total-box';
+import { useAppStore } from '@notional-finance/notionable-hooks';
+interface DataSectionProps {
+  title: string;
+  button?: {
+    label: string;
+    icon?: React.ReactNode;
+    onClick?: () => void;
+    externalLink?: string;
+  };
+  totalBoxes: {
+    title: string;
+    value: number;
+    decimals?: number;
+    suffix?: string;
+    prefix?: string;
+    trend?: {
+      value: number;
+      direction: 'up' | 'down';
+      duration: string;
+    };
+  }[];
+  children?: React.ReactNode;
+}
+
+// Data Section Component
+const DataSection = ({
+  title,
+  button,
+  totalBoxes,
+  children,
+}: DataSectionProps) => {
+  const { isMobileView } = useAppStore();
+  const theme = useTheme();
+  return (
+    <DataSectionContainer>
+      <HeaderContainer>
+        {isMobileView ? (
+          <LargeInputTextEmphasized>{title}</LargeInputTextEmphasized>
+        ) : (
+          <H2>{title}</H2>
+        )}
+        {button && !isMobileView && (
+          <Button
+            variant="contained"
+            sx={{
+              backgroundColor: alpha(theme.palette.primary.main, 0.1),
+              color: theme.palette.primary.main,
+            }}
+            startIcon={button.icon}
+            endIcon={
+              button.externalLink && <ExternalLinkIcon sx={{ fontSize: 16 }} />
+            }
+            onClick={button.onClick}
+            href={button.externalLink}
+          >
+            {button?.label}
+          </Button>
+        )}
+      </HeaderContainer>
+      <ContentContainer>
+        <TotalBoxContainer>
+          {totalBoxes.map((totalBox) => (
```

Comment:
I think we should just use children here and make the `TotalBoxContainer` and `TotalBoxes` reusable components we can pass in as children. There will be sections that do not  have any total boxes.